### DISCUSSION TITLE
Startup Race Condition

### DISCUSSION THREAD

**Discussion by victoraugustolls** [fastapi/fastapi#7890] [Questions] [answered]
### First check

* [X] I used the GitHub search to find a similar issue and didn't find it.
* [X] I searched the FastAPI documentation, with the integrated search.
* [X] I already searched in Google "How to X in FastAPI" and didn't find any information.

### Description
Is this the best practice? It seems to be causing some sort of race condition that causes a timeout on the loop when connecting to the database. Code below:
```

import logging
import os
import sys

from fastapi import FastAPI

from app.dependencies import set_authenticator, set_database, get_database
from app.middlewares import (
    NewRelicMiddleware,
    OpenCensusMiddleware,
    SentryMiddleware,
)
from app.repositories import Cache, PostgresRepository
from app.routers import salesforce_router
from app.salesforce import Authenticator

app = FastAPI()


@app.on_event("startup")
async def startup():
    try:
        db = await PostgresRepository.create()
        set_database(db)

        authenticator = Authenticator(
            cache=Cache(),
            instance=os.environ.get("AUTH_INSTANCE"),
            version=46.0,
        )
        set_authenticator(authenticator=authenticator)
    except Exception as e:
        print(f"Error initializing database: {e}")
        sys.exit(1)


@app.on_event("shutdown")
async def shutdown():
    logging.info("Closing database connections")
    db = get_database()
    await db.close()


app.add_middleware(SentryMiddleware)
app.add_middleware(NewRelicMiddleware)
app.add_middleware(OpenCensusMiddleware)

app.include_router(auth_router)
```

**Comment by dmontagu** [comment]
I don't see anything obviously wrong here, but it's not totally clear to me what part of the code you are asking about? It's also not clear to me what you mean by "the loop". Is there a traceback / error message of some sort?

Are you sure you are successfully able to connect to your database?

**Comment by victoraugustolls** [comment]
Yes I am! When running locally everything runs fine. When I deploy to an Azure Webapp, I start so receive errors but they seem like a loop errors. Will collect the trace and put them here!

**Comment by victoraugustolls** [comment, marked as answer]
Find out to be a bad configuration on Azure firewall, but as the connection couldn’t be accomplished, the cpu started to go crazy and some weird traces that looked like race conditions started to appear. Thanks for the help!

**Comment by tiangolo** [comment]
Thanks for the help here @dmontagu! :cake: :bowing_man: 

And thanks @victoraugustolls for reporting back and closing the issue :+1:
